build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/ElasticsearchJavaBasePlugin.java

```diff
@@ -196,6 +197,49 @@ private static void configureNativeLibraryPath(Project project) {
         });
     }
 
+    private static void configureEntitlements(Project project) {
+        Configuration agentJarConfig = project.getConfigurations().create("entitlementAgentJar");
+        Project agent = project.findProject(":libs:entitlement:agent");
+        if (agent != null) {
+            agentJarConfig.defaultDependencies(deps -> {
+                deps.add(project.getDependencies().project(Map.of("path", ":libs:entitlement:agent", "configuration", "default")));
+            });
+        }
+        FileCollection agentFiles = agentJarConfig;
+
+        Configuration bridgeJarConfig = project.getConfigurations().create("entitlementBridgeJar");
+        Project bridge = project.findProject(":libs:entitlement:bridge");
```

Comment:
our binary plugin implementations should not know anything of our gradle project structure. Any dependency declarations should live in the build scripts or in a convention plugin. None of this logic IMO belongs in the java base plugin as the java base plugin does only deal with java base buidling and also should not handle test configurations.

Reply:
I think all of this here should for now at least be be part of the ElasticsearchTestBase plugin as its all about dealing with the test setup. Ultimately we want those project dependencies to be moved out from here into a convention plugin. We (I can take care of that in a folllow up)

Regarding